nixos/modules/services/web-apps/reposilite.nix

```diff
@@ -0,0 +1,253 @@
+    mkPackageOption
+    mkOption
+    mkIf
+    getExe
+    types
+    optionals
+    ;
+
+  # Reposilite actually uses https://github.com/dzikoysk/cdn, but its format is similar enough to YAMLs to use pkgs.formats.yaml.
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  options.services.reposilite = {
+    enable = mkEnableOption "reposilite";
+    package = mkPackageOption pkgs "reposilite" { };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          hostname = mkOption {
+            description = ''
+              Hostname
+              The hostname can be used to limit which connections are accepted.
+              Use 0.0.0.0 to accept connections from anywhere.
+              127.0.0.1 will only allow connections from localhost.
+            '';
+            default = "0.0.0.0";
+            example = "127.0.0.1";
+            type = types.str;
+          };
+
+          port = mkOption {
+            description = "Port to bind";
+            default = 8080;
+            example = 5678;
+            type = types.port;
+          };
+
+          database = mkOption {
+            description = ''
+              Database configuration. Supported storage providers:
+              - mysql localhost:3306 database user password
+              - sqlite reposilite.db
+              - sqlite --temporary
+              Experimental providers (not covered with tests):
+              - postgresql localhost:5432 database user password
+              - h2 reposilite
+            '';
+            default = "sqlite reposilite.db";
+            example = "mysql localhost:3306 database user password";
+            type = types.str;
+          };
+
+          sslEnabled = mkOption {
+            description = "Support encrypted connections";
+            default = false;
+            type = types.bool;
+          };
+
+          sslPort = mkOption {
+            description = "SSL port to bind";
+            default = 443;
+            example = 5678;
+            type = types.port;
+          };
+
+          keyPath = mkOption {
+            description = ''
+              Key file to use.
+              You can specify absolute path to the given file or use ''${WORKING_DIRECTORY} variable.
+              If you want to use .pem certificate you need to specify its path next to the key path.
+              Example .pem paths setup:
+              keyPath = "''${WORKING_DIRECTORY}/cert.pem ''${WORKING_DIRECTORY}/key.pem"
+              Example .jks path setup:
+              keyPath = "''${WORKING_DIRECTORY}/keystore.jks"
+            '';
+            default = "\${WORKING_DIRECTORY}/cert.pem \${WORKING_DIRECTORY}/key.pem";
+            example = "\${WORKING_DIRECTORY}/keystore.jks";
+          };
+
+          keyPassword = mkOption {
+            description = "Key password to use";
+            default = "";
+            type = types.str;
+          };
```

Comment:
Same here about credentials ending up in the nix store

Reply:
What's the best way to go about fixing this, since the program just reads it in plaintext from the config file?